Sources/Mustache/Lambda.swift

```diff
@@ -34,7 +34,7 @@
 ///
 public struct MustacheLambda {
     /// lambda callback
-    public typealias Callback = (Any, MustacheTemplate) -> String
+    public typealias Callback = (String) -> Any?
```

Comment:
The lambda loses the parameter's possible type info here, that's suboptimal.
Not sure if it's properly possible to not lose the type info. I can see how it can be hard.
I'm not sure what was the previous behavior but it would still be nice to not lose the type info.

Reply:
This is the difficulty of trying to follow the lead of various dynamically typed languages. I could instead return a String and forget supporting some of the more esoteric options like returning lambdas from lambdas.

Reply:
Can we wrap this `String` in a wrapper type?

```swift
public struct LambdaInvocation {
  ...
}
```

I'd start with one property (`var parameter: String`), allowing the flexibility to potentially enrich it in the future

Reply:
@Joannis It is only ever initialised with a String. The spec is pretty clear 
> When any value found during the lookup is a callable object, such as a function or lambda, the object will be invoked and passed the block of text. The text passed is the literal block, unrendered.

I looked at just returning a String and doesn't work. The spec allows for you to return an object, to be dereferenced. Even the example is the manual does this.

> Template:
> ```
> * {{time.hour}}
> ```
> Context:
> ```
> {
>   "time": function() {
>     return {
>       "hour": 8,
>       "minute": 0,
>       "second": 0
>     }
>   }
> }
> ```
> Output:
> ```
> * 8
> ```